iree/compiler/Dialect/LinalgExt/IR/TiledOpInterface.cpp

```diff
@@ -238,6 +238,59 @@ struct InsertSliceTiledOpInterface
     return extractSliceOp;
   }
 };
+
+/// Forwards the implementation of `TiledOpInterface` to upstream
+/// `TilingInterface`. Note that this forwarding is only valid when the
+/// iteration space is same as the data space of the result(s). This is due to
+/// the difference in the tiling algorithm being developed around
+/// `TilingInterface` and that used with `TiledOpInterface`. The difference
+/// comes down to the former only needing the tiled operation, and not the value
+/// of the whole tensor.
+template <typename OpTy>
+struct ForwardToTilingInterface
+    : public TiledOpInterface::ExternalModel<ForwardToTilingInterface<OpTy>,
+                                             OpTy> {
+  SmallVector<Value> getDestinationOperands(Operation *op, OpBuilder &b) const {
+    return cast<OpTy>(op).getDestinationOperands(b);
+  }
+
+  SmallVector<StringRef> getLoopIteratorTypes(Operation *op) const {
+    return cast<OpTy>(op).getLoopIteratorTypes();
+  }
+  SmallVector<Range> getLoopBounds(Operation *op, OpBuilder &b) const {
+    return cast<OpTy>(op).getLoopBounds(b);
+  }
+  Operation *getTiledImplementation(Operation *op, OpBuilder &b,
+                                    ValueRange dest,
+                                    ArrayRef<OpFoldResult> offsets,
+                                    ArrayRef<OpFoldResult> sizes,
+                                    SmallVectorImpl<Value> &results) const {
+    Operation *tiledOp =
+        cast<OpTy>(op).getTiledImplementation(b, dest, offsets, sizes);
+    Location loc = op->getLoc();
+    if (op->getNumResults() != dest.size()) {
+      op->emitOpError(
```

Comment:
I didn't check the tiling interface API in details. Is the result always expected to be valid? Can we return nullptr to abort tiling? 

Also this check can happen at the beginning?

Reply:
Oh good catch! Changed it. Returning a `nullptr` is indeed an error indication. So added a check for that and fixed this check as well (s/`op`/`tiledOp`)